UnitsNet/UnitConverter.cs

```diff
@@ -384,8 +393,9 @@ private static bool HasParameterTypes(MethodInfo methodInfo, params Type[] expec
         private static bool TryParseUnit(Type unitType, string unitName, out object unitValue)
         {
             unitValue = null;
-
-            if(!Enum.IsDefined(unitType, unitName))
+            var eNames = Enum.GetNames(unitType);
+            unitName = eNames.FirstOrDefault(x => x.Equals(unitName, StringComparison.OrdinalIgnoreCase));
```

Comment:
We should probably use Enum.TryParse which includes a bool parameter to specify ignoring case. See [here](https://docs.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=netframework-4.7.2#System_Enum_TryParse__1_System_String_System_Boolean___0__).

Reply:
This would also prefer this method, but I can't use that here without using reflection, which I was trying to avoid. The try parse method you linked takes a generic type argument, which means that the type has to be known at compile type. We have a Type parameter known only at run time. Is there another TryParse that you know of that takes the enum type "Type" as parameter?

Reply:
`Enum.TryParse<T>(string, bool, out T)` exists, but I suspect since we target .NET 4.0 is why we can't use it.